api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -453,6 +453,40 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  message LocalReplyConfig {
+    // Configuration of list of mappers which allows to filter and change local response. 
+    // Code iterate through mappers until first match of filter.
+    repeated ResponseMapper mapper = 1;
+
+    // Allows to define custom format of local reply.
+    oneof reply_format {
+      // Response body plain text format.
+      string format = 2;
+  
+      // Response body json format as dictionary.
+      google.protobuf.Struct json_format = 3;
+    }
+  };
+
+  // Configuration of local reply returned by Envoy. Allows to specify mapping and format of response.
+  LocalReplyConfig local_reply_config = 35;
```

Comment:
Should `LocalReplyConfig` be one-per `ResponseMapper`? What's the tradeoff here?

Reply:
I think `LocalReplyConfig` should be one by `http_conn_manager`, but we can define many Mappers. The tradeoff is that you can define only one format per manager. So if you want to define custom format for different response it won't be possible.

Reply:
If we think that there might be possibly a use case here for multiple custom format, I'd just allow it in the API; the implementation can probably deal with this without too much complexity. I.e. let's be really sure that one format will be sufficient forever if we take the existing design.

Reply:
I added possibility to define format for each mapper. I think about implementation: if mapper format is defined then use mapper format else use format from config (or default if not defined)

Reply:
Given the way you added it, it seems we could punt this until later, if you really want to keep the top-level `reply_format`. So, I'd either rollback that change or remove the top-level `reply_format`; I gather you want it though to reduce verbosity if you have many rewriters?

Reply:
Yes I thought it might be good idea to define one format at top-level and allow to define custom one for each mapper if someone want different format. I can also remove top level mapper but if someone want to change the format of all responses it will require definition of mapper without `filter/rewriter`.

Reply:
OK, given that, maybe just completely ignore the above thread and rollback the per-rewriter bits, since they can be added in a non-breaking way later additively. It'll simplify the implementation work at your end. Sorry about the confusion, the PR clarified how this would look, thanks.